woocommerce/woocommerce-ios
# Add-ons: Add Variations Attributes row

closes #4312
fixes #3702

# Why

As part of the Variations changes, we are trying to make the variation flow simpler. This PR takes out the "Product Variation Attributes" section that was inside the "Variations List" screen and makes it accessible from the "Product Form" screen.

# How

- Update `ProductAttribute` to conform to `GeneratedCopiable`

- Update `ProductFormActionsFactory` to include the `.attributes` actions for the variable products when the product has at least one attribute.

- Update `DefaultProductFormTableViewModel` to properly render the `.attributes` row and update the `.variations` row to show the variations count.

- Update `ProductFormViewController` to navigate to the "Edit Attributes" screen after the new "Variation Attributes" row is tapped.

- Update `ProductVariationsViewController` to remove the ability to edit product attributes for that screen.


# Demo

https://user-images.githubusercontent.com/562080/120580636-275df080-c3ef-11eb-8fc2-4f7d68d51694.mov

# Testing Steps

- Launch the app and navigate to a product with variations

- See that there is a "Variations" row stating how many variations the product has.

- See that there is a "Variations Attributes" row.

- Tap on the "Variations Attributes" row and see that it navigates to the "Edit Attributes" screen.

- Edit/Add/Delete an attribute

- Go back the to the main product screen, see that the "Variations Attribute" row is updated.



Update release notes:
